sdk/src/azure/core/az_log.c

```diff
@@ -19,8 +19,38 @@
 static az_log_classification const* volatile _az_log_classifications = NULL;
 static az_log_message_fn volatile _az_log_message_callback = NULL;
 
+// Verifies that the classification that the user provided is one of the valid possibilties and
+// guards against looping past the end of the classification array.
+// Make sure to update the switch statement whenever new classifications are added.
+static bool _az_is_valid_log_classification(az_log_classification const* classifications)
```

Comment:
The name should be plural, because this is what it does. I.e. `_az_is_valid_log_classifications`, or, better, I think, - `_az_log_classifications_are_valid`